### ISSUE TITLE
Move from transifex to Weblate

### ISSUE THREAD

**Issue by return42** [searxng/searxng#195]
Over the years, transifex has become proprietary crap.

We should move to a [self-hosted](https://weblate.org/de/download/) weblate [1] instance I think.

Related-to: https://github.com/searx/searx/issues/268#issuecomment-769630687
Discussed-at: https://github.com/searx/searx/discussions/2552

[1] https://github.com/WeblateOrg/weblate

**Comment by dalf** [comment]
The [hardware requirement](https://docs.weblate.org/en/latest/admin/install/venv-debian.html) is:
* 2 GB of RAM (4GB recommended)
* 2 CPU cores 
* 1 GB of storage space

Do you have any suggestion where to host this service?

**Comment by return42** [comment]
> The hardware requirement is:

I think the the hardware requirement is not my problem since we will host only one project with a handful of contributors ("concurrent users", see C&P below).

> Do you have any suggestion where to host this service?

The work I see is maintaining the service, managing accounts and having a backup strategy, a vacation replacement of the admin .. etc ... it is the organizational tasks that still make me hesitate.

If it make sense I can start a _"pilot"_ on my darmarit.org server .. later if we think it is a value solution we can consider to rent some hardware and own a domain name that fits (I can pay from my company).  My Limitation: at the moment I have a lot to do for my customers and do not know yet when I will have time for a _"pilot"_  ... suggestions?

----

![grafik](https://user-images.githubusercontent.com/554536/126658210-f836a2a6-7b20-47e9-a151-307bb40fd5e8.png)

**Comment by comradekingu** [comment]
@bourrel Use https://hosted.weblate.org/hosting/ or https://weblate.bubu1.eu/
Add the user "kingu" as admin.

**Comment by return42** [comment]
@comradekingu not sure I understand you, do you know a weblate hoster open for FLOSS projects like SearXNG?

**Comment by comradekingu** [comment]
@return42 Both of those are exactly that.

**Comment by return42** [comment]
@dalf what do you think about creating a project at 

- https://hosted.weblate.org/hosting/ 
- or https://weblate.bubu1.eu/ 

I do not have any preferences / both support registration via github account.  May be `hosted.weblate.org` has a wider user base. The  `weblate.bubu1.eu` service is hosted by Marcus Hoffmann, see https://bubu1.eu/services/

**Comment by comradekingu** [comment]
I'd say Bubu. Guy is doing a lot to help with decentralization of various ecosystems.
Two good options.

**Comment by dalf** [comment]
I have just try bubu1.

There is this error when I try to create a new component "from version control":
![image](https://user-images.githubusercontent.com/1594191/127126709-3eb1d13e-b708-4af5-8211-ddea0f4d9d1b.png)
...

![image](https://user-images.githubusercontent.com/1594191/127126581-57613054-5f83-4065-8a85-52679097260d.png)

**Comment by return42** [comment]
I am a bit confused by all these Chinese language tags. In `searx/translations` we have

- zh_CN
- zh_TW added in 62690f62 `pull translations from Transifex` / Sun Jan 6 11:26:03 2019
- zh_Hant_TW:  b70ce84eb added in  `[enh] update translations - add taiwanese and filipino` /  Sun Feb 18 23:42:30 2018

I think the existing zh_Hant_TW is crap:

1. In the zh_Hant_TW po-file the language is tagged by `"Language: zh_TW\n"`
2. zh_Hant_TW has never been updated since creation, zh_TW has been updated regularly.

If I have understand everything right:

- `zh_Hant_??` is _traditional_ in region `??` (CN, TW , ..) and 
- `zh_Hans_??` is _simplified_  in region  `??`(CN, TW , ..) and  ~~(is the same for all regions / no region suffix required)~~
- `zh_Hant` traditional Chinese
- `zh_Hans` simplified Chinese
-  zh_CN, zh_TW, zh_HK .. [are old and should be replaced](https://public-inbox.org/libc-alpha/s9dblumvzp3.fsf@taka.site/T/) by `zh_Hant_??`

To summarize up: I think we have to

1. delete existing `zh_Hant_TW`
2. rename `zh_TW` into `zh_Hant_TW`
3. rename `zh_CN` into `zh_Hant_CN`

**update** I mixed the commit hash values .. now it is correct

**Comment by comradekingu** [comment]
[In Remmina](https://hosted.weblate.org/projects/remmina/remmina/) it is
zh_Hans for simplified
zh_Hant for traditional
with no complaints

Weblate 4.7.2 additonally has 
zh_Hans_SG
zh_Hant_HK
zh_Latn
Nothing for TW, but that might be in error.

Filemask looks to be correct.

**Comment by return42** [comment]
> Nothing for TW, but that might be in error.

I don't hink so, I'm pretty sure it's because of `zh_Hant_TW` and `zh_TW` existing in our translation tree .. both are the same by definition, both are  traditional Chinese .. and this is what the error message says.

From where did you get the information that there is _"Nothing for TW"_? .. I can't find a list of language tags in the Weblate documentation, I assume Weblate [supports RFC 5646](https://docs.weblate.org/en/latest/faq.html#why-does-weblate-use-language-codes-such-sr-latn-or-zh-hant)

From [RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646.html#section-2.2)

    The namespace of language tags and their subtags is administered by
    the Internet Assigned Numbers Authority (IANA)

In the [IANA registry language-subtag-registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) `zh-Hant-TW` is registered:

    %%
    Type: redundant
    Tag: zh-Hant-TW
    Description: Taiwan Chinese in traditional script
    Added: 2005-04-11

IANA uses hyphen while POSIX style using underscore as a separator

> Filemask looks to be correct.

otherwise weblate would not have found duplicates  ;-)

**Comment by comradekingu** [comment]
You can't have more than one for the same thing, or unspecified ones. That is where the ambiguity comes from.

The available options are in the list of possible new translations to start (minus existing ones), or between
https://github.com/WeblateOrg/language-data/blob/main/weblate_language_data/language_codes.py
and
https://github.com/WeblateOrg/language-data/blob/main/weblate_language_data/locale/en_GB/LC_MESSAGES/django.po
The actual preferences are, somewhere.
Edit: 
https://github.com/WeblateOrg/language-data/blob/main/languages.csv has both
Edit2: Here we go
https://github.com/WeblateOrg/language-data/blob/main/aliases.csv
or
https://github.com/WeblateOrg/language-data/blob/main/weblate_language_data/aliases.py

**Comment by dalf** [comment]
> 1. delete existing zh_Hant_TW
> 1. rename zh_TW into zh_Hant_TW
> 1. rename zh_CN into zh_Hant_CN

It seems the solution (*) according to 

* https://github.com/flask-admin/flask-admin/issues/403 (from 2014...)
* and
```ipython
In [2]: babel.Locale.parse("zh_TW")
Out[2]: Locale('zh', territory='TW', script='Hant')

In [3]: babel.Locale.parse("zh_Hant_TW")
Out[3]: Locale('zh', territory='TW', script='Hant')

In [4]: babel.Locale.parse("zh_CN")
Out[4]: Locale('zh', territory='CN', script='Hans')

In [5]: babel.Locale.parse("zh_Hans_CN")
Out[5]: Locale('zh', territory='CN', script='Hans')
```

(*) but it requires [to change settings.yml for all instances](https://github.com/searxng/searxng/blob/master/searx/settings.yml#L1659-L1704)...

--> The minimal solution is to delete zh_Hant_TW (the translation that has not been updated).

ping @MarcAbonce

**Comment by comradekingu** [comment]
[RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646.html) according to https://docs.weblate.org/en/latest/faq.html#why-does-weblate-use-language-codes-such-sr-latn-or-zh-hant :)

**Comment by MarcAbonce** [comment]
From an old PR comment I made a while ago: https://github.com/searx/searx/pull/2531

> This [the PR] assumes that our UI locales have nothing more than language and optionally a country.
If we ever add a script specific locale like zh_Hant_TW* this would have to change to accommodate that, but the idea would be pretty much the same as this fix.
> 
> *By the way, in our Transifex page it looks like zh_Hant_TW was removed (I assume it was merged with zh_TW which is already in traditional Chinese). Should we remove this directory or am I making a wrong assumption here?

Just to confirm: yeah, the `zh_Hant_TW` was redundant and never used.

But it's worth pointing out that most of our locale handling assumes that locales are in a `/language(_region)?/` format and are not always tested for `/language_script(_region)?/` locales.

So if for some reason you want to support `language_script` variants, it may not work out of the box.

**Comment by dalf** [comment]
So I suggest to keep zh_CN, zh_TW for now.

---

![image](https://user-images.githubusercontent.com/1594191/127290388-1ef2d834-a251-411e-ba0a-8337e88bb0c9.png)

I guess the "Monolingual base language file" should be the .pot file.
Does that mean that the .pot should be committed in the git repository ?

**Comment by comradekingu** [comment]
#231 can be fixed in
https://weblate.bubu1.eu/settings/searxng/searxng/ and
https://weblate.bubu1.eu/settings/searxng/glossary/

**Comment by return42** [comment]
> Edit2: Here we go
> https://github.com/WeblateOrg/language-data/blob/main/aliases.csv
> or
> https://github.com/WeblateOrg/language-data/blob/main/weblate_language_data/aliases.py

just want to leave a link here / see: https://github.com/WeblateOrg/language-data/issues/303#issuecomment-895216072

and https://github.com/WeblateOrg/language-data/commit/d06e654994b5d7fa9d2ce2145e71b5a59d55ff07

**Comment by return42** [comment]
@unixfox do you speak flemish (dutch)? 

If so, do you like to translate https://weblate.bubu1.eu/projects/searxng/searxng/nl_BE/ ?

// sorry for being unknown, I do not know if flemish is a dutch dialect or not.  If I am not wrong, in belgium there are two languages .. one is a dutch dialect (used in north) and the other part (in the south) of belgium speaks french ... I'm wrong?

~**Edit:** Oops, now I see there is also a traditional language in the south https://weblate.bubu1.eu/projects/searxng/searxng/cy/~

**Comment by unixfox** [comment]
> @unixfox do you speak flemish (dutch)? 
> 
> If so, do you like to translate https://weblate.bubu1.eu/projects/searxng/searxng/nl_BE/ ?
> 
> // sorry for being unknown, I do not know if flemish is a dutch dialect or not.  If I am not wrong, in belgium there are two languages .. one is a dutch dialect (used in north) and the other part (in the south) of belgium speaks french ... I'm wrong?
> 
> ~**Edit:** Oops, now I see there is also a traditional language in the south https://weblate.bubu1.eu/projects/searxng/searxng/cy/~

We speak three languages in Belgium: German, Dutch and French.

Unfortunately I'm from wallonia so I only speak French even though I learnt some dutch at school but I stopped learning it so I can't translate in dutch anymore sorry.
